src/components/OmouComponents/IdleLogout.js

```diff
@@ -0,0 +1,178 @@
+
+    const handleLogout = useCallback(() => {
+        console.log("logging out...")
+        closeRegistrationCart();
+        dispatch(logout());
+        history.push("/login");
+    }, [dispatch, history]);
+
+    let timeout = 1200000; //1080000;
+    const modalTimeout = 300000 //120000;
+    const [remaining, setRemaining] = useState(timeout);
+    const [elapsed, setElapsed] = useState(0);
+    const elapseDiff = 5000;
+
+    const [lastActive, setLastActive] = useState(+new Date());
+    const [isIdle, setIsIdle] = useState(false);
+
+    const handleOnActive = () => setIsIdle(false);
+    const handleOnIdle = () => setIsIdle(true);
+
+    const classes = useStyles();
+    // getModalStyle is not a pure function, we roll the style only on the first render
+    const [modalStyle] = useState(getModalStyle);
+    const [open, setOpen] = useState(false);
+
+   
+    function logoutAfter2Minutes() {
+        return new Promise(resolve => {
+            setTimeout(() => {
+                resolve(logoutAndCloseModal());
+            }, 300000);
+        })
+    };
+    
+    const logoutAndCloseModal = () => {
+        handleClose();
+        handleLogout();
+    }
+
+    async function handleOpen() {
+        setOpen(true);
+        handleReset();
+        setRemaining(modalTimeout);
+        await logoutAfter2Minutes();
+
+    };
+    
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    const {
+        reset,
+        pause,
+        resume,
+        getRemainingTime,
+        getLastActiveTime,
+        getElapsedTime
+    } = useIdleTimer({
+        timeout,
+        onActive: handleOnActive,
+        onIdle: handleOnIdle
+    });
+    
+    const handleReset = () => reset();
+    const handlePause = () => pause();
+    const handleResume = () => resume();
+
+    useEffect(() => {
+        setRemaining(getRemainingTime())
+        setLastActive(getLastActiveTime())
+        setElapsed(getElapsedTime())
+    
+        setInterval(() => {
+            setRemaining(getRemainingTime())
+            setLastActive(getLastActiveTime())
+            setElapsed(getElapsedTime())
+        }, 1000);
+    }, []);
+
+    const body = () => {
+        return (
+            <div style={modalStyle} 
+                className={classes.Idle}
+                data-cy="activityCheckModal">
+                <p id="simple-modal-description">
+                    <Typography variant="h5" className={classes.IdleFont}>Are you still there?</Typography>
+                    <div style={{"text-align": "center"}}>
+                        <Button 
+                            onClick={handleClose}
+                            className={classes.YesButton}
+                            data-cy="activityModalSubmit"
+                        >Yes</Button>
+                    </div>
+                </p>
+            </div>
+        )};
+    return (
+        <div>
+            {
+                (remaining === 0) && handleOpen()
+            }
```

Comment:
change 161 - 163 to 1 line

Reply:
does "remaining" mean remaining seconds? milliseconds? make the variable name more specific